File: backend/app/schemas/repo_integration.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.repo_integration import RepoProvider


class RepoIntegrationCreate(BaseModel):
    repo_url: str = Field(min_length=1, max_length=2048)


class RepoIntegrationRead(BaseModel):
    id: int
    project_id: int
    provider: RepoProvider
    repo_owner: str
    repo_name: str
    repo_url: str
    default_branch: str | None
    is_connected: bool
    last_verified_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/repoIntegration.ts
export type RepoProvider = "github";

export interface RepoIntegration {
  id: number;
  project_id: number;
  provider: RepoProvider;
  repo_owner: string;
  repo_name: string;
  repo_url: string;
  default_branch: string | null;
  is_connected: boolean;
  github_repository_id?: number | null;
  github_installation_id?: number | null;
  is_private?: boolean;
  connection_mode?: "public_url" | "github_app";
  last_verified_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface RepoIntegrationInput {
  repo_url: string;
}
